tasks/sign-index-image/sign-index-image.yaml

```diff
@@ -4,117 +4,86 @@ kind: Task
+      default: "data.json"
+    - name: request
       type: string
-      description: targetIndex signing image
-    - name: requestUpdateTimeout
+      description: Signing pipeline name to handle this request
+      default: "hacbs-signing-pipeline"
+    - name: referenceImage
       type: string
-      default: "360"
-      description: Max seconds waiting for the status update
-    - name: signingPipelineImage
+      description: The image to be signed.
+    - name: manifestDigestImage
       type: string
-      default: "quay.io/redhat-isv/operator-pipelines-images:released"
-      description: An image with CLI tools needed for the signing
+      default: ""
+      description: Manifest Digest Image used to extract the SHA
     - name: requester
       type: string
-      value: ""
-      description: Name of the user that requested the signing, for auditing purposes
-    - name: configMapName
+      description: The request user
+    - name: requestTimeout
       type: string
-      description: ConfigMap Name for this request
-    - name: pipelineRunName
-      type: string
-      description: The name of the Parent PipelineRun of this task
-  results:
-    - name: requestMessage
-      type: string
-      description: Internal Request message
-    - name: requestReason
-      type: string
-      description: Internal Request reason
-    - name: requestResults
-      type: string
-      description: Internal Request results
+      default: "180"
+      description: InternalRequest timeout
+  workspaces:
+    - name: data
+      description: workspace to read and save files
   steps:
     - name: sign-index-image
       image:
-        quay.io/hacbs-release/release-utils@sha256:5733ece907aa70d6ebced36484c936731e8b27bfcf87fed226a0ecf95047a6b8
+        quay.io/hacbs-release/release-utils:6e92a6f8df8ef1cbecfb4c25b73ec6d92bded527
       script: |
         #!/usr/bin/env sh
-        PATH=/bin:/usr/bin:/usr/local/bin
-        export PATH
+        #
+        set -e
 
-        jsonInputFile="/tmp/$$.json"
-        cat > ${jsonInputFile} <<JSON
-        $(params.requestJsonResults)
-        JSON
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
 
-        internalIndexImageCopyResolved=`jq -cr .jsonBuildInfo ${jsonInputFile} \
-        | jq -cr .internal_index_image_copy_resolved`
+        # The signing configuration should be set as data.sign in the releaseAdmissionPlan. Previously it used to be
+        # set also in the data.fbc. The data should be set in the ReleaseAdmissionPlan as follows:
+        #
+        # data:
+        #   sign:
+        #     request: <signing pipeline name>
+        #     pipelineImage: <image-pullspec>
+        #     configMapName: <configmap name>
+        #
+        request=$(jq -r '.sign.request // .fbc.request // "$(params.request)"' ${DATA_FILE})
```

Comment:
Does this work? It looks like it has two fallback defaults? I know we have had troubles in the past with variables in jq defaults

Reply:
It works. But this might be unnecessary. I wanted to keep some compatibility with the current format definition, of having the signing information in `data.fbc`. But maybe instead we should make sure the RPA is changed to the proper way.

Reply:
BTW, regarding "troubles in the past with variables in jq defaults" - in this case there is no shell variable at play. In this case it's a tekton context variable which is substituted with the value before the script is called. So for these `$(...)` variables it doesn't really matter where they are placed - in `''` or `""` or anywhere else - they will always be expanded before the script is called.